script.js

```diff
@@ -0,0 +1,908 @@
+let currAmtWon = 0;
+// Track payout level separate from creditsInserted
+let payOutLevel = 0;
+
+// Hardcoded individual credit payouts for X amount of credits used
+const oneCreditPayOut = [250, 50, 25, 9, 6, 4, 3, 2, 1, 0];
+const twoCreditPayout = [500, 100, 50, 18, 12, 8, 6, 4, 2, 0];
+const threeCreditPayout = [750, 150, 75, 27, 18, 12, 9, 6, 3, 0];
+const fourCreditPayout = [1000, 200, 100, 36, 24, 16, 12, 8, 4, 0];
+const fiveCreditPayout = [4000, 250, 125, 45, 30, 20, 15, 10, 5, 0];
+
+const payOutSchedule = [[...oneCreditPayOut], [...twoCreditPayout],
+  [...threeCreditPayout], [...fourCreditPayout], [...fiveCreditPayout]];
+
+// ----------------- Animation Management --------------//
+// Tracks new card images when swapped
+let newCardImage;
+// Tracks cardImages during new dealt
+let cardImage;
+// Tracks cover-card animations;
+let coverCardImage;
+let coverCardShown = true;
+
+// Track Rank of Hand
+// Five of a kind being 0 and Jacks or Better being 8, no winning hand = 9;
+let rankOfHand = 9;
+
+// --- Scroll Display Management ----//
+let combiDisplayCol;
+let pOut1Col;
+let pOut2Col;
+let pOut3Col;
+let pOut4Col;
+let pOut5Col;
+let combiDisplayMachine;
+let pOut1Machine;
+let pOut2Machine;
+let pOut3Machine;
+let pOut4Machine;
+let pOut5Machine;
+
+// -- Game Round/State Management --- //
+let gameRound = 0;
+// States are between 'startGame'/ 'playerNextMove'/ 'gameOver'
+let gameState = 'startGame';
+
+//= ================== Helper Functions =========================//
+
+// Function that determines payoutLevel
+const calcPayOutLevel = () => {
+  if (numCreditsInserted > 0) {
+    payOutLevel = numCreditsInserted - 1;
+    console.log(payOutLevel, 'payoutlevel');
+  }
+};
+
+// Calculate the score of the hand and add to user's credits
+const calcHandScore = () => {
+  const amtWon = payOutSchedule[payOutLevel][rankOfHand];
+  currAmtWon += amtWon;
+  console.log(amtWon, 'amtWon');
+  return currAmtWon;
+};
+
+// Function that calculates the rank of the current hand
+const compareCardRank = () => {
+  let pointer = 0;
+  let numOfCommonCards = 0;
+  const numRemainingCards = playerHand.length;
+  while (pointer < playerHand.length - 1) {
+    let j = pointer + 1;
+    while (j < playerHand.length) {
+      if (playerHand[pointer].rank === playerHand[j].rank) {
+        numOfCommonCards += 1;
+        commonCardsArray.push(playerHand[pointer]);
+      }
+      j += 1;
+    }
+    numCommonCardsArray.push(numOfCommonCards);
+    numOfCommonCards = 0;
+    pointer += 1;
+  }
+};
+
+// CheckForKinds scenarios checked
+//                       Written    Tested
+// Five of a kind          x          x
+// Straight Flush          x          x
+// Four of a kind          x          x
+// Full House              x          x
+// Flush                   x          x
+// Straight                x          x
+// Three of a kind         x          x
+// Two Pair                x          x
+// One Pair(J or Better)   x          x
```

Comment:
Great checklist for development! Can move this to some project management document after development because the code may not be the best place to store project completion checklists. Otherwise all of our code would just be littered with checklists. 

Code is helpful for comments explaining what's included in the code though, e.g. if you wanted to list the types of hand types that this code checks for. That's helpful and relevant for future readers.

Reply:
Noted, I was thinking of moving it to a readme doc, but haven't had time to clean it up